test/window-integration-tests/test.js

```diff
@@ -0,0 +1,96 @@
+const Application = require("spectron").Application;
+const electronPath = require("electron");
+const libnut = require("../..");
+const { POS_X, POS_Y, WIDTH, HEIGTH, TITLE } = require("./constants");
+
+let app;
+const APP_TIMEOUT = 10000;
+
+beforeAll(async () => {
+    app = new Application({
+        path: electronPath,
+        args: ['main.js'],
+        startTimeout: APP_TIMEOUT,
+        waitTimeout: APP_TIMEOUT,
+    });
+    await app.start();
+    await app.client.waitUntilWindowLoaded();
+    await app.browserWindow.minimize();
+    await app.browserWindow.restore();
+    await app.browserWindow.focus();
+});
+
+describe("getWindows", () => {
+    it("should list our started application window", () => {
+        // GIVEN
+
+        // WHEN
+        const windowNames = libnut.getWindows().map(handle => libnut.getWindowTitle(handle));
+
+        // THEN
+        expect(windowNames).toContain(TITLE);
+    });
+});
+
+describe("getActiveWindow", () => {
+    it("should return our started application window", () => {
+        // GIVEN
+
+        // WHEN
+        const activeWindowHandle = libnut.getActiveWindow();
```

Comment:
This is more like a `GIVEN` state, isn't it? Same goes for the other tests. But this is just my opinion. Nothing to seriously argue about.

Reply:
I've been thinking about this as well, but in the end I went with `WHEN`.
The `getActiveWindow` just didn't feel like a `GIVEN` to me, can't really explain why, though 😄